File: solutions/0001-two-sum/solution.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();

        int[] result = new int[2];
        
        for (int j = 0; j < nums.length; j++) {
            int difference = target - nums[j];

            if(map.containsKey(difference)) {
                result[0] = map.get(difference);
                result[1] = j;
                break;
            }

            map.put(nums[j], j);
        }

        return result; 

    }
}

File: solutions/0001-two-sum/solution.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """

        table = {}
        res = []
        for i in range(len(nums)):
            difference = target - nums[i]

            if difference in table:
                return [table[difference], i]
        
            
            table[nums[i]] = i

        return []

            
        
